extension/app/ts/simulation/services/SimulationModeEthereumClientService.ts

```diff
@@ -471,7 +485,7 @@ export class SimulationModeEthereumClientService {
 		}
 		if('fromBlock' in logFilter) {
 			const logs = await this.ethereumClientService.getLogs(logFilter)
-			if(logFilter.toBlock === 'latest' || logFilter.toBlock === await this.getBlockNumber() ) {
+			if (logFilter.toBlock === 'latest' || logFilter.toBlock > await this.ethereumClientService.getBlockNumber() ) {
 				return [...logs, ...await this.getSimulatedLogs(logFilter)]
 			}
 			return logs
```

Comment:
This feels racey.  Can we instead do something like:
```suggestion
			const logs = await this.ethereumClientService.getLogs(logFilter)
			const [simulatedBlockNumber, simulatedLogs] = await this.getSimulatedLogs(logFilter)
			if (logFilter.toBlock === 'latest' || logFilter.toBlock >= simulatedBlockNumber) {
				return [...logs, ...await simulatedLogs]
			} else {
				return logs
			}
```
This would ensure that we aren't getting logs for a different "block number" than the block number we test against to see if we *should* get logs.